### DISCUSSION TITLE
how to implement textobject all pairs similar to targets in mini.ai?

### DISCUSSION THREAD

**Discussion by zoomlin** [nvim-mini/mini.nvim#1172] [Q&A] [answered]
autocmd User targets#mappings#user call targets#mappings#extend({
    \ 's': { 'separator': [{'d':','}, {'d':'.'}, {'d':';'}, {'d':':'}, {'d':'+'}, {'d':'-'},
    \                      {'d':'='}, {'d':'~'}, {'d':'_'}, {'d':'*'}, {'d':'#'}, {'d':'/'},
    \                      {'d':'\'}, {'d':'|'}, {'d':'&'}, {'d':'$'}] },
    \ '@': {
        \     'separator': [],
    \     'pair':      [{'o':'(', 'c':')'}, {'o':'[', 'c':']'}, {'o':'{', 'c':'}'}, {'o':'<', 'c':'>'}],
    \     'quote':     [{'d':"'"}, {'d':'"'}, {'d':'`'}],
    \     'tag':       [{}],
    \     },
    \ })

**Comment by echasnovski** [comment]
I am not sure what this is supposed to do. Maybe take a look at [example](https://github.com/echasnovski/mini.nvim/blob/0bd6c4d25f2b0cc1ecb8b1a0f659cce54627e218/doc/mini-ai.txt#L335-L360) and there will somehting helpful there.

**Comment by zoomlin** [comment]
I need one text object for quotes and brackets to do vi@, to make selection between quotes or brackets with one shortcut, from example i don't understand how to join two text object in one

**Reply by echasnovski** [reply, marked as answer]
Try this one:

```lua
require('mini.ai').setup({
  custom_textobjects = {
    ['@'] = { { '%b()', '%b[]', '%b{}', '%b""', "%b''", '%b``' }, '^.().*().$' },
  },
})
```

This looks fairly similar to [the first example](https://github.com/echasnovski/mini.nvim/blob/0bd6c4d25f2b0cc1ecb8b1a0f659cce54627e218/doc/mini-ai.txt#L335-L336) from the link. And 'mini.ai' allows [balanced pattern for same left and right characters](https://github.com/echasnovski/mini.nvim/blob/0bd6c4d25f2b0cc1ecb8b1a0f659cce54627e218/doc/mini-ai.txt#L397).
